chore(docs): AS-13 docs rewrite (#263)
voxel51/fiftyone-teams-app-deploy: helm/docs/upgrading.md

# Upgrading FiftyOne Teams

Voxel51 assumes you use the published
Helm Chart to deploy your FiftyOne Teams environment.
If you are using a custom deployment
mechanism, carefully review the changes in the
[Helm Chart](https://github.com/voxel51/fiftyone-teams-app-deploy)

> and update your deployment accordingly.

Author: Moved

   a new kubernetes secret:

    ```shell
    kubectl --namespace your-namepace-here create secret generic \
        fiftyone-license --from-file=license=./your-license-file
    ```

1. [Upgrade to FiftyOne Teams version 2.1.3](#upgrading-from-previous-versions)
1. Upgrade FiftyOne Teams SDK users to FiftyOne Teams version 2.1.3
    - Login to the FiftyOne Teams UI
    - To obtain the CLI command to install the FiftyOne SDK associated with
      your FiftyOne Teams version, navigate to `Account > Install FiftyOne`
1. Upgrade all the datasets

    > **NOTE** Any FiftyOne SDK less than 2.1.3 will lose connectivity after
    > this point.
    > Upgrading all SDKs to `fiftyone==2.1.3` is recommended before migrating
        > your database.

    ```shell
    FIFTYONE_DATABASE_ADMIN=true fiftyone migrate --all
    ```

1. Validate that all datasets are now at version 0.25.1

    ```shell
    fiftyone migrate --info
    ```

### From FiftyOne Teams Version 2.0.0

1. [Upgrade to FiftyOne Teams version 2.1.3](#upgrading-from-previous-versions)
1. Voxel51 recommends upgrading all FiftyOne Teams SDK users to FiftyOne Teams

>    version 2.1.3, but it is not required

Reviewer: ```suggestion
   version 2.2.0, but it is not required
```

>     - Login to the FiftyOne Teams UI
>     - To obtain the CLI command to install the FiftyOne SDK associated with

Reviewer: ```suggestion
    1. Login to the FiftyOne Teams UI
    1. To obtain the CLI command to install the FiftyOne SDK associated with
```

      your FiftyOne Teams version, navigate to `Account > Install FiftyOne`
1. Voxel51 recommends that you upgrade all your datasets, but it is not
   required.  Users using the FiftyOne Teams 2.0.0 SDK will continue to operate
   uninterrupted during, and after, this migration

   ```shell
   FIFTYONE_DATABASE_ADMIN=true fiftyone migrate --all
   ```

1. To ensure that all datasets are now at version 0.25.1, run

   ```shell
   fiftyone migrate --info
   ```

### From FiftyOne Teams Version 2.1.3

1. [Upgrade to FiftyOne Teams version 2.2.0](#upgrading-from-previous-versions)
1. Voxel51 recommends upgrading all FiftyOne Teams SDK users to FiftyOne Teams
   version 2.2.0, but it is not required

>     - Login to the FiftyOne Teams UI
>     - To obtain the CLI command to install the FiftyOne SDK associated with

Reviewer: ```suggestion
    1. Login to the FiftyOne Teams UI
    1. To obtain the CLI command to install the FiftyOne SDK associated with
```

      your FiftyOne Teams version, navigate to `Account > Install FiftyOne`
1. Voxel51 recommends that you upgrade all your datasets, but it is not
   required.  Users using the FiftyOne Teams 2.0.0 SDK will continue to operate
   uninterrupted during, and after, this migration

   ```shell
   FIFTYONE_DATABASE_ADMIN=true fiftyone migrate --all
   ```

1. To ensure that all datasets are now at version 1.1.0, run

   ```shell
   fiftyone migrate --info
   ```

#### FiftyOne Teams v2.2+ Delegated Operator Changes

FiftyOne Teams v2.2 introduces some changes to delegated operators, detailed
below.

##### Delegated Operation Capacity

> By default, all deployments are provisioned with capacity to support up to 3

Reviewer: ```suggestion
By default, all deployments are provisioned with capacity to support up to three
```

# Upgrading FiftyOne Teams

<!-- toc -->

- [Upgrading From Previous Versions](#upgrading-from-previous-versions)
  - [From Early Adopter Versions (Versions less than 1.0)](#from-early-adopter-versions-versions-less-than-10)
  - [From Before FiftyOne Teams Version 1.1.0](#from-before-fiftyone-teams-version-110)
  - [From FiftyOne Teams Versions After 1.1.0 and Before Version 1.6.0](#from-fiftyone-teams-versions-after-110-and-before-version-160)
  - [From FiftyOne Teams Versions 1.6.0 to 1.7.1](#from-fiftyone-teams-versions-160-to-171)
  - [From FiftyOne Teams Version 2.0.0](#from-fiftyone-teams-version-200)
  - [From FiftyOne Teams Version 2.1.3](#from-fiftyone-teams-version-213)
    - [FiftyOne Teams v2.2+ Delegated Operator Changes](#fiftyone-teams-v22-delegated-operator-changes)
      - [Delegated Operation Capacity](#delegated-operation-capacity)

>       - [Existing Orchestrators](#existing-orchestrators)

Reviewer: What do you think about reversing the order? More relevant/recent up front? That's less scrolling or interaction from the user.

Author: I don't think it matters much with a TOC (just use the link), but I can change the order.
